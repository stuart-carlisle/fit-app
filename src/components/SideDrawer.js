import React from 'react'
import SideDrawerItem from './SideDrawerItem'
import SideDrawerBackButton from './SideDrawerBackButton'
import { history } from '../routers/AppRouter'
import SideDrawerLogoutButton from './SideDrawerLogoutButton';

export const SideDrawer = ({ drawerToggleClickHandler, show }) => {
    const link = history.location.pathname
    
    let sideDrawerClasses
    

    if (show){
        sideDrawerClasses= "side-drawer open"
    }
    
    const sideDrawerBackData = {
        image: "/images/back-arrow-white.svg",
        name: "BACK",
        link: link,
    }
    const sideDrawerArray = [
    {
        image: "/images/home-icon.svg",
        name: "DASHBOARD",
        link: "/dashboard"
    },{
        image: "/images/weight-icon.svg",
        name: "SUBMIT WEIGHT",
        link:"/personal-details"
    },{
        image: "/images/target-icon.svg",
        name:"TARGETS",
        link:"/update-targets"
    },{
        image: "/images/progress-icon.svg",
        name:"PROGRESS TRACKER",
        link:"/progress"
    },{
        image: "/images/diary-icon.svg",
        name:"DIARY",
        link:"/diary"
    }]
    const sideDrawerLogOutData = {
        name: "LOGOUT",
        link:"/"
    }

    return (
    <div className={sideDrawerClasses}>
        <SideDrawerBackButton data={sideDrawerBackData} drawerToggleClickHandler={drawerToggleClickHandler} />
        {(  
            sideDrawerArray.map((sideDrawerItemData) => {
               return <SideDrawerItem key={sideDrawerItemData.name} data={sideDrawerItemData} drawerToggleClickHandler={drawerToggleClickHandler} />
            })
        )}
        <SideDrawerLogoutButton data={sideDrawerLogOutData} drawerToggleClickHandler={drawerToggleClickHandler} />
    </div>
   ) 

} 

export default SideDrawer