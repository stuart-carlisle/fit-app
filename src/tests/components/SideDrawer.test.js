import { shallow } from 'enzyme'
import React from 'react'
import SideDrawer from '../../components/SideDrawer'
 
test('Should render the sidedrawer correctly',()=>{
    const drawerToggleClickHandler = jest.fn()
    const wrapper = shallow(<SideDrawer show={true} drawerToggleClickHandler={drawerToggleClickHandler} />)
    expect(wrapper).toMatchSnapshot()
})