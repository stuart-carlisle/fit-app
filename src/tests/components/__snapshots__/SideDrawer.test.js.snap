// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the sidedrawer correctly 1`] = `
<div
  className="side-drawer open"
>
  <SideDrawerBackButton
    data={
      Object {
        "image": "/images/back-arrow-white.svg",
        "link": "blank",
        "name": "BACK",
      }
    }
    drawerToggleClickHandler={[Function]}
  />
  <SideDrawerItem
    data={
      Object {
        "image": "/images/home-icon.svg",
        "link": "/dashboard",
        "name": "DASHBOARD",
      }
    }
    drawerToggleClickHandler={[Function]}
    key="DASHBOARD"
  />
  <SideDrawerItem
    data={
      Object {
        "image": "/images/weight-icon.svg",
        "link": "/personal-details",
        "name": "SUBMIT WEIGHT",
      }
    }
    drawerToggleClickHandler={[Function]}
    key="SUBMIT WEIGHT"
  />
  <SideDrawerItem
    data={
      Object {
        "image": "/images/target-icon.svg",
        "link": "/update-targets",
        "name": "TARGETS",
      }
    }
    drawerToggleClickHandler={[Function]}
    key="TARGETS"
  />
  <SideDrawerItem
    data={
      Object {
        "image": "/images/progress-icon.svg",
        "link": "/progress",
        "name": "PROGRESS TRACKER",
      }
    }
    drawerToggleClickHandler={[Function]}
    key="PROGRESS TRACKER"
  />
  <SideDrawerItem
    data={
      Object {
        "image": "/images/diary-icon.svg",
        "link": "/diary",
        "name": "DIARY",
      }
    }
    drawerToggleClickHandler={[Function]}
    key="DIARY"
  />
  <Connect(SideDrawerLogoutButton)
    data={
      Object {
        "link": "/",
        "name": "LOGOUT",
      }
    }
    drawerToggleClickHandler={[Function]}
  />
</div>
`;
